.github/workflows/deployment.yaml

```diff
@@ -30,3 +30,11 @@ jobs:
             npm i
           fi
           yarn deploy
+      - name: Generate Seneca-IPC144.pdf
+        run: |
+          chmod 777 ./convert-to-pdf.sh
+          yarn to-pdf
+          mv -f Seneca-IPC144.pdf ./static
+          git add ./static/Seneca-IPC144.pdf
```

Comment:
We shouldn't need to do this.  Much as the final HTML we generate doesn't get stored in the `main` branch, neither should this.  It should be a build/deploy artifact only, which gets sent to gh-pages.

Reply:
I made a change as your request, however, when I tested on my local branch and tried downloading, it failed and it says there are no files in the ./static folder

Reply:
You need to run the build before doing the pdf stuff, so that it exists.  Change the order of operations.

Reply:
HI @humphd ,

Currently, the `convert-to-pdf.sh` is generating `pdf` from `https://ipc144.sdds.ca/`. In order to get the updated `.pdf` the page should be deployed before doing the pdf generation and finally the page should be deployed again so that it includes the updated pdf.

1. yarn build 
2. yarn deploy
3. Convert to pdf from the deployed page: yarn to-pdf
4. Move `pdf` to `static` folder: mv -f ./Seneca-IPC144.pdf ./static
5. Deploy again to update the new pdf: yarn deploy

I tested this on my local branch and it works. However, deploying two times seems to be inefficient, I came up with another approach which is to create a local server on GH-Actions. 

1. yarn build
2. Create a local server on Github Actions to generate pdf:
`yarn serve & npx wait-on http://0.0.0.1:3000/`
3. Change `convert-to-pdf.sh` from `https://ipc144.sdds.ca/` to `http://0.0.0.1:3000`. Then convert pdf from local server: yarn to-pdf
4. Move `pdf` to `static` folder: mv -f ./Seneca-IPC144.pdf ./static
5. yarn deploy

However, I have difficulty in creating a local server on GH-Actions I'm not sure if this solution can work.

Maybe there're better approaches, would like to know your opinion @humphd

Reply:
cc @TueNguyen2911, who can help us understand the PDF bits.  NOTE: we already run a server as part of the PDF generation scripts, see https://github.com/Seneca-ICTOER/IPC144/blob/main/convert-to-pdf.js.  All you should need to do is `build` the site, then this will do the rest.

Reply:
Hi @trangntt-016, if you run `yarn serve-convert`, it will build the pages and run [convert-to-pdf.js](https://github.com/Seneca-ICTOER/IPC144/blob/main/convert-to-pdf.js). In `convert-to-pdf.js`, it firstly will serve the static files generated in `build` to `localhost:3000` and then run `convert-to-pdf.sh` to convert to pdf. There's no need for a `yarn deploy`

Reply:
I would do the steps like this: 
1. yarn serve-convert
2. mv -f ./Seneca-IPC144.pdf ./static
3. yarn deploy again to update the link to download the pdf